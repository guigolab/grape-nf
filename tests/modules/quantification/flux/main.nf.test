nextflow_process {

    name "Test Process quantify"
    script "modules/quantification/flux/main.nf"
    process "quantify"

    tag "module"

    test("Should quantify with Flux Capacitor (uncompressed annotation)") {

        when {
            process {
                """
                input[0] = file("${baseDir}/data/annotation.gtf")
                input[1] = Channel.from([
                    [ "sample3", "test3", file("${baseDir}/data/sample3_m4_n10_toGenome.bam"), "bam", "GenomeAlignments", true, "MATE2_SENSE" ]
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("Should quantify with Flux Capacitor (compressed annotation)") {

        when {
            process {
                """
                input[0] = file("${baseDir}/data/annotation.gtf.gz")
                input[1] = Channel.from([
                    [ "sample3", "test3", file("${baseDir}/data/sample3_m4_n10_toGenome.bam"), "bam", "GenomeAlignments", true, "MATE2_SENSE" ]
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
