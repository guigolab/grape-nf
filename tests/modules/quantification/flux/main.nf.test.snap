{
    "Should quantify with Flux Capacitor (uncompressed annotation)": {
        "content": [
            {
                "0": [
                    [
                        "sample3",
                        "test3",
                        "sample3.isoforms.gtf:md5,1a039c489fdeb3e0ab4ff768424b5997",
                        "gtf",
                        "TranscriptQuantifications",
                        true,
                        "MATE2_SENSE"
                    ]
                ],
                "1": [
                    [
                        "sample3",
                        "test3",
                        "sample3.genes.gff:md5,74f9de8e31ff2c24bc45af5f8ec73530",
                        "gtf",
                        "GeneQuantifications",
                        true,
                        "MATE2_SENSE"
                    ]
                ],
                "genes": [
                    [
                        "sample3",
                        "test3",
                        "sample3.genes.gff:md5,74f9de8e31ff2c24bc45af5f8ec73530",
                        "gtf",
                        "GeneQuantifications",
                        true,
                        "MATE2_SENSE"
                    ]
                ],
                "isoforms": [
                    [
                        "sample3",
                        "test3",
                        "sample3.isoforms.gtf:md5,1a039c489fdeb3e0ab4ff768424b5997",
                        "gtf",
                        "TranscriptQuantifications",
                        true,
                        "MATE2_SENSE"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-30T17:41:10.263296"
    },
    "Should quantify with Flux Capacitor (compressed annotation)": {
        "content": [
            {
                "0": [
                    [
                        "sample3",
                        "test3",
                        "sample3.isoforms.gtf:md5,1a039c489fdeb3e0ab4ff768424b5997",
                        "gtf",
                        "TranscriptQuantifications",
                        true,
                        "MATE2_SENSE"
                    ]
                ],
                "1": [
                    [
                        "sample3",
                        "test3",
                        "sample3.genes.gff:md5,74f9de8e31ff2c24bc45af5f8ec73530",
                        "gtf",
                        "GeneQuantifications",
                        true,
                        "MATE2_SENSE"
                    ]
                ],
                "genes": [
                    [
                        "sample3",
                        "test3",
                        "sample3.genes.gff:md5,74f9de8e31ff2c24bc45af5f8ec73530",
                        "gtf",
                        "GeneQuantifications",
                        true,
                        "MATE2_SENSE"
                    ]
                ],
                "isoforms": [
                    [
                        "sample3",
                        "test3",
                        "sample3.isoforms.gtf:md5,1a039c489fdeb3e0ab4ff768424b5997",
                        "gtf",
                        "TranscriptQuantifications",
                        true,
                        "MATE2_SENSE"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-30T17:30:38.512548"
    }
}